### ISSUE TITLE
The outout of `glimpse()` is too wide for RStudio in Visual Markdown mode

### ISSUE THREAD

**Issue by GegznaV** [r-lib/pillar#656]
In RStudio Visual Markdown mode, the output of `glimpse()` is distorted (the summary of one column does not fit into a single line)

Steps to reproduce the issue:
1) select a data frame with a big enough number of rows;
2) apply `glimpse()` on the data frame.

Current output of `pillar::glimpse(iris)`:
![image](https://github.com/r-lib/pillar/assets/12725868/aad855a1-3004-442f-b71c-5ab1ffa3f4c4)

Expected output (the output in console):
![image](https://github.com/r-lib/pillar/assets/12725868/a6d32d82-6fbe-48ec-a5c8-d1e85fa8ab81)

If the editor's pane is made even wider, the issue gets even worse:
![image](https://github.com/r-lib/pillar/assets/12725868/335b07f9-cee2-4431-85c0-b76bf9824d3f)

**Comment by krlmlr** [comment]
Thanks. The width is controlled by `getOption("width")` , IIRC. Can you check the value of this option in the visual markdown editor? Can you work around by setting the option to a smaller value?

**Comment by GegznaV** [comment]
Yes, sure. In the following circumstances, the width is 133:

![image](https://github.com/r-lib/pillar/assets/12725868/43e69fe0-bd43-4456-879e-09071fd5da16)

So I'm not sure if this is an issue with Visual Editor, which provides the wrong value of output width, or of `glimpse()`, which does not correct it appropriately. What do you think?

**Comment by krlmlr** [comment]
It seems that the `glimpse()` output is at most 132 characters per line?

In the IDE, the `"width"` option is changed when you resize the console tab. Does this have an effect on the visual editor?

**Comment by GegznaV** [comment]
Yes, the value changes. But it seems that `getOption("width")` shows the total width of the editor (or console) window and not the width of the Visual Editor output area. The same width when I run code in Visual editor and in the console:

![image](https://github.com/r-lib/pillar/assets/12725868/b43257ad-01f4-4f07-b1e7-56d5bc57cc0d)

**Comment by GegznaV** [comment]
One more test. I measured the number of characters that fit into a single line of output:

![image](https://github.com/r-lib/pillar/assets/12725868/ea82561b-f982-4bbe-8521-41f9b7028736)

**Comment by krlmlr** [comment]
Can you please take this to https://github.com/rstudio/rstudio then? This will also affect other packages that use the `"width"` option. This might be a known issue, haven't looked.
